docs/internal/release-versions.md

```diff
@@ -0,0 +1,50 @@
+# Release Versions
+
+## Introduction
+
+We have several pieces of software:
+
+* Chain Core itself
+* SDKs (currently Java, with Ruby and Node.js in the roadmap)
+* Mac app
+* Windows installer
+* Docker container
+
+Each of these is going to be updated for one of three reasons:
+
+* its own improvements (e.g., a bugfix in the SDK or a Mac app)
+* improvements in a product as a whole (e.g. a new feature in Chain Core with updated SDK support)
+* as a dependency on another piece (e.g. fix in a Chain Core needs corresponding Mac and Windows executables)
+
+We need a consistent, yet simple, versioning scheme that allows both individual updates as well as grouped updates: from major feature releases to smaller, maintenance releases.
+
+We probably do not want to formally commit to specific semantics of the version numbers, such as [SemVer](http://semver.org), but we need to keep all numbers in order, so it’s easier navigate them.
+
+## Proposal
+
+Use the following version format:
+
+    <major>.<minor>.<build>
```

Comment:
@tessr Since some may expect not to experience breaking changes from 1.5 to 1.6, we should try to keep it that way. And if we need to break something without major version bump, then we'll make sure we have proper migrations or deprecation notices, or translation layers in place. Probably the doc should expand on that to make it more clear.